sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ParallelTransferOptions.java

```diff
@@ -5,48 +5,44 @@
 
 import com.azure.core.annotation.Fluent;
 import com.azure.storage.blob.ProgressReceiver;
+import com.azure.storage.blob.specialized.BlockBlobAsyncClient;
 import com.azure.storage.common.implementation.Constants;
 import com.azure.storage.common.implementation.StorageImplUtils;
 
+import static com.azure.storage.blob.BlobAsyncClient.BLOB_DEFAULT_NUMBER_OF_BUFFERS;
+import static com.azure.storage.blob.BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE;
+
 /**
  * This class contains configuration used to parallelize data transfer operations.
  */
 @Fluent
-public class ParallelTransferOptions {
-
-    private static final int BLOB_DEFAULT_UPLOAD_BLOCK_SIZE = 4 * Constants.MB;
-    private static final int BLOB_MAX_BLOCK_SIZE = 100 * Constants.MB;
+public final class ParallelTransferOptions {
 
-    private static final int BLOB_DEFAULT_NUMBER_OF_PARALLEL_TRANSFERS = 8;
+    private static final int BLOB_MAX_UPLOAD_BLOCK_SIZE = 100 * Constants.MB;
 
-    private int blockSize;
-    private int numBuffers;
+    private Integer blockSize;
+    private Integer numBuffers;
     private ProgressReceiver progressReceiver;
 
     /**
      * Creates a new {@link ParallelTransferOptions} with default parameters applied.
-     * blockSize = 4MB
-     * numBuffers = 8
      */
```

Comment:
Thoughts on changing this class from a Fluent to a constructor based design? Given its simplicity that might make it easier to use and could remove the need for the apply defaults API.

`public ParallelTransferOptions()` -> Uses base line defaults
`public ParallelTransferOptions(Integer, Integer)` -> Uses what is passed, otherwise defaults (most common scenario)
`public ParallelTransferOptions(Integer, Integer, ProgressReceiver)` -> Same as above but can take a progress reported

Reply:
I don't know what the guidelines are on when to use Fluent and when to use constructors, so I'm fine with this if you think it's best

Reply:
Yeah, I think this should be constructor based and immutable. For any operation, the set of options is consistent if this is immutable.